extensions/amp-date-countdown/0.1/amp-date-countdown.js

```diff
@@ -0,0 +1,283 @@
+      this.win.clearInterval(this.countDownTimer_);
+    }
+    this.renderItems_(Object.assign(data, this.localeWordList_));
+  }
+
+  /**
+   * @return {number}
+   * @private
+   */
+  getEpoch_() {
+    let epoch;
+
+    if (this.endDate_) {
+      epoch = Date.parse(this.endDate_);
+    } else if (this.timestampMs_) {
+      epoch = this.timestampMs_;
+    } else if (this.timestampSeconds_) {
+      epoch = this.timestampSeconds_ * 1000;
+    }
+
+    user().assert(!isNaN(epoch),
+        'One of end-date, timestamp-ms, timestamp-seconds is required');
+
+    return epoch;
+  }
+
+  /**
+   * @param {string} locale
+   * @return {Object}
+   * @private
+   */
+  getLocaleWord_(locale) {
+    if (LOCALE_WORD[locale]) {
+      const localeWordList = LOCALE_WORD[locale];
+      return {
+        'years': localeWordList[0],
+        'months': localeWordList[1],
+        'days': localeWordList[2],
+        'hours': localeWordList[3],
+        'minutes': localeWordList[4],
+        'seconds': localeWordList[5],
+      };
+    } else {
+      user().error(TAG, `Invalid locale '${locale}', return empty locale word`);
+      return {};
+    }
+  }
+
+  /**
+   * @param {number} ms
+   * @return {Object}
+   * @private
+   */
+  getYDHMSFromMs_(ms) {
+
+    const priorityList = {
+      days: 1,
+      hours: 2,
+      minutes: 3,
+      seconds: 4,
+    };
+
+    const d = priorityList[this.biggestUnit_] == 1
+      ? (Math.trunc((ms) / MILLISECONDS_IN_DAY))
+      : 0;
+    const h = priorityList[this.biggestUnit_] == 2
+      ? (Math.trunc((ms) / MILLISECONDS_IN_HOUR))
+      : priorityList[this.biggestUnit_] < 2
+        ? (Math.trunc((ms % MILLISECONDS_IN_DAY) / MILLISECONDS_IN_HOUR))
+        : 0;
+    const m = priorityList[this.biggestUnit_] == 3
+      ? (Math.trunc((ms) / MILLISECONDS_IN_MINUTE))
+      : priorityList[this.biggestUnit_] < 3
+        ? (Math.trunc((ms % MILLISECONDS_IN_HOUR) / MILLISECONDS_IN_MINUTE))
+        : 0;
+    const s = priorityList[this.biggestUnit_] == 4
+      ? (Math.trunc((ms) / MILLISECONDS_IN_SECOND))
+      : (Math.trunc((ms % MILLISECONDS_IN_MINUTE) / MILLISECONDS_IN_SECOND));
+
+    return {
+      d,
+      dd: this.padStart_(d),
+      h,
+      hh: this.padStart_(h),
+      m,
+      mm: this.padStart_(m),
+      s,
+      ss: this.padStart_(s),
+    };
+  }
+
+  /**
+   * @param {number} input
+   * @return {Object}
+   * @private
+   */
+  padStart_(input) {
+    if (input < -9 || input > 9) {
+      return String(input);
+    } else if (input >= -9 && input < 0) {
+      return '-0' + Math.abs(input);
+    }
+
+    return '0' + input;
+  }
+
+  /**
+   * @param {!Element} element
+   * @private
+   */
+  rendered_(element) {
+    const template = this.element.firstElementChild;
+    removeChildren(this.element);
+    this.element.appendChild(template);
+    this.element.appendChild(element);
+
+    const event = createCustomEvent(this.win,
```

Comment:
This should not be needed. We normally trigger `AmpEvents.DOM_UPDATE` when new amp-components are added/removed from DOM (like in amp-list)